mathml/presentation-markup/mrow/mrow-fallback-002.html

```diff
@@ -0,0 +1,151 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>Test mrow fallback for some MathML elements (preferred logical widths)</title>
+<link rel="help" href="https://w3c.github.io/mathml-core/#script-and-limit-schemata">
+<meta name="assert" content="Verify that invalid markup fallbacks to mrow behaviour when computing preferred logical widths.">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/mathml/support/feature-detection.js"></script>
+<script src="/mathml/support/fonts.js"></script>
+<script src="/mathml/support/layout-comparison.js"></script>
+<script src="/mathml/support/mathml-fragments.js"></script>
+<style>
+  /* Revert style specified in the UA style sheet that changes box size. */ 
+  mfrac { padding-inline: 0; }
+</style>
+<script>
+  setup({ explicit_done: true });
+  window.addEventListener("load", () => { loadAllFonts().then(runTests); });
+
+  function isValid(tagName, mspaceCount) {

```

Comment:
I think this is duplicating a lot of code from `mrow-fallback-002.html`. Can you please factor out some code into a `mathml/presentation-markup/mrow/support/invalid-markup.js` file? I guess that would be `isValid()` and the code generating the invalid markup.

Reply:
Done